settings.gradle.kts

```diff
@@ -28,12 +28,12 @@ rootProject.name = "maven-plugin-development"
 
 includeBuild("sample/gradle-producer-build")
 
-if (System.getenv("CI") == "true") {
-    gradleEnterprise {
-        buildScan {
-            publishAlways()
-            termsOfServiceUrl = "https://gradle.com/terms-of-service"
-            termsOfServiceAgree = "yes"
-        }
+develocity {
+    buildScan {
+        termsOfUseUrl = "https://gradle.com/terms-of-service"
+        termsOfUseAgree = "yes"
+        // TODO: workaround for https://github.com/gradle/gradle/issues/22879.
+        val isCI = providers.environmentVariable("CI").isPresent
+        publishing.onlyIf { isCI }
```

Comment:
How about this?
```suggestion
        publishing.onlyIf { System.getenv("CI") }
```

Reply:
`providers.environmentVariable` is the preferred API on Gradle side, but I can pick yours if you want.

Reply:
Why do you think it's the preferred AP?

Reply:
As it's CC compatible and lazy.

Reply:
Using native APIs directly is also CC compatible because access to these APIs will include the value in the configuration cache. You're right that using `providers.environmentVaribale` has the benefit of being lazy. There can be cases where you can avoid adding the variable to the configuration cache key this way. For example when you pass the variable to some task as an input, but that task is not configured for that particular build execution. In our case, `publishing.onlyIf { }` is lazy, but this will be evaluated for every build anyway. So, in this particular case, I would prefer the simplicity of using the native API.

Reference: https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:reading_sys_props_and_env_vars